src/mortality.jl

```diff
@@ -0,0 +1,195 @@
+) where {R,W}
+LoglogisticMortalityR{R,W}(hillcoefficient, timestep, nothing)
+end
+
+precalcrule(rule::LoglogisticMortalityR, data) = precalc_timestep(rule, data)
+
+@inline function applyrule(data, rule::LoglogisticMortalityR, (N, X, R), I) # RETURN ONLY POPULATION
+    N > zero(N) || return zero(N)
+    β = get(data, rule.hillcoefficient, I...)
+    @fastmath N * R^β / (R^β + X^β)
+end
+
+
+"""
+    ExponentialMortality(; rate, threshold, timestep)
+    ExponentialMortality{R}(; rate, threshold, timestep)
+    ExponentialMortality{R,W}(; rate, threshold, timestep)
+
+Exponential mortality based on exposure threshold and mortality rate parameter, using exact solution.
+
+Exponential mortality based on exposure grid ``X``, an exposure threshold parameter ``z``
+    and a mortality rate ``r`` using exact solution between time ``t`` and ```t+1``:
+
+```math
+N_{t+1} = N_{t}e^{-r t*(X-z)}
+```
+
+# Keyword Arguments
+
+- `rate`: Mortality rate.
+- `threshold`: Exposure threshold under which there is no effect.
+- `timestep`: Time step for the growth rate, in a type compatible with the simulation `tspan`.
+
+`rate` and `threshold` can be a `Number`, an [`Aux`](@ref) array or another [`Grid`](@ref).
+
+Pass grid `Symbol`s to `R` or both `R` and `W` type parameters to use to specific grids.
+Read: Tuple(population, exposure)
+Written: Tuple(population)
+"""
+struct ExponentialMortality{R,W,MR,ZX,TS,S} <: GrowthRule{R,W}
+    rate::MR
+    threshold::ZX
+    timestep::TS
+    nsteps::S
+end
+function ExponentialMortality{R,W}(; 
+    rate=INTRINSICRATE_PARAM,
+    threshold=THRESHOLD_EXPOSURE,
+    timestep, 
+) where {R,W}
+    ExponentialMortality{R,W}(rate, threshold, timestep, nothing)
+end
+
+precalcrule(rule::ExponentialMortality, data) = precalc_timestep(rule, data)
+
+@inline function applyrule(data, rule::ExponentialMortality, (N,X), I)
+    N > zero(N) || return zero(N)
+    rt = get(data, rule.rate, I...) * rule.nsteps
+    z = get(data, rule.threshold, I...) 
+    return @fastmath N * exp(-rt*max(0,X-z))
+end
+
+"""
+    ExponentialMortalityR(; threshold, timestep)
+    ExponentialMortalityR{R}(; threshold, timestep)
+    ExponentialMortalityR{R,W}(; threshold, timestep)
+
+Exponential mortality based on exposure grid ``X``, an exposure threshold parameter ``z``
+    and a grid-based mortality rate ``r``.
+Rate is given by another grid reflecting change according to resistance mechanisms.
+
+```math
+N_{t+1} = N_{t}e^{-r t*(X-z)}
+```
+
+# Keyword Arguments
+
+- `threshold`: Exposure threshold under which there is no effect.
+- `timestep`: Time step for the growth rate, in a type compatible with the simulation `tspan`.
+
+`rate` and `threshold` can be a `Number`, an [`Aux`](@ref) array or another [`Grid`](@ref).
+
+Pass grid `Symbol`s to `R` or both `R` and `W` type parameters to use to specific grids.
+Read: Tuple(population, exposure, rate)
+Written: Tuple(population)
+"""
+struct ExponentialMortalityR{R,W,ZX,TS,S} <: GrowthRule{R,W}
```

Comment:
Yes, And I can even simplified the others by replacing the Tuple `(population, exposure)` to only `population` and use `Grid(:exposure)` and `Grid(:rate)` !

Reply:
Only use `Grid(:exposure)` if you see it as a parameter. It's easier to optimise grid access when they're in the type signature.

It should only be used by users in scripts, never inside a rule (as we don't know the grid names in a rule) - it would always have with the option of using `Aux` or a single value insteaad. But if that's what you mean then yes :)